### ISSUE TITLE
Option "git-config-email" being empty has no effect

### ISSUE THREAD

**Issue by hasezoey** [JamesIves/github-pages-deploy-action#868]
## Describe the bug

Also, setting option `git-config-email` to be explicitly empty, has no effect (still using original email) (this should either be mentioned or behave like when doing `git commit --author "Someone <>"`)

## Reproduction Steps

Use action like in the `Getting Started `, have the mentioned option set to be empty (see below for the full file) and have a commit that triggers this workflow with email

## Logs

Workflow link: https://github.com/typegoose/typegoose/runs/3575932754 (Step name `Deploy to Github Pages`)

<details>
<summary>Logs</summary>

```txt
Run JamesIves/github-pages-deploy-action@4.1.5

    GitHub Pages Deploy Action 🚀

    🚀 Getting Started Guide: https://github.com/marketplace/actions/deploy-to-github-pages
    ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
    🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues

    📣 Maintained by James Ives: https://jamesiv.es
    💖 Support: https://github.com/sponsors/JamesIves
Checking configuration and starting deployment… 🚦
Deploying using Deploy Token… 🔑
Configuring git…
/usr/bin/git config user.name github-pages-deploy
/usr/bin/git config user.email hasezoey@gmail.com
/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
/usr/bin/git remote rm origin
/usr/bin/git remote add origin ***github.com/typegoose/typegoose.git
Git configured… 🔧
Starting to commit changes…
/usr/bin/git ls-remote --heads ***github.com/typegoose/typegoose.git refs/heads/gh-pages
0c8313f253b51096552528d12ed891f914e95794	refs/heads/gh-pages
Creating worktree…
/usr/bin/git fetch --no-recurse-submodules --depth=1 origin gh-pages
From https://github.com/typegoose/typegoose
 * branch            gh-pages   -> FETCH_HEAD
 * [new branch]      gh-pages   -> origin/gh-pages
/usr/bin/git worktree add --no-checkout --detach github-pages-deploy-action-temp-deployment-folder
Preparing worktree (detached HEAD 4be0230)
/usr/bin/git checkout -B gh-pages origin/gh-pages
Previous HEAD position was 4be0230 docs(hooks): add "options" parameter
Switched to a new branch 'gh-pages'
Branch 'gh-pages' set up to track remote branch 'gh-pages' from 'origin'.
/usr/bin/rsync -q -av --checksum --progress /home/runner/work/typegoose/typegoose/. github-pages-deploy-action-temp-deployment-folder --delete --exclude CNAME --exclude .ssh --exclude .git --exclude .github --exclude github-pages-deploy-action-temp-deployment-folder
Checking if there are files to commit…
/usr/bin/git add --all .
/usr/bin/git checkout -b github-pages-deploy-action/m5icb7ze8
Switched to a new branch 'github-pages-deploy-action/m5icb7ze8'
/usr/bin/git commit -m Deploying to gh-pages from @ typegoose/typegoose@4be023058b352fde261f7ebb6a6a02788d8a877e 🚀 --quiet --no-verify
/usr/bin/git push --force ***github.com/typegoose/typegoose.git github-pages-deploy-action/m5icb7ze8:gh-pages
To https://github.com/typegoose/typegoose.git
   0c8313f..1750a72  github-pages-deploy-action/m5icb7ze8 -> gh-pages
Changes committed to the gh-pages branch… 📦
Running post deployment cleanup jobs… 🗑️
/usr/bin/git checkout -B github-pages-deploy-action/m5icb7ze8
Reset branch 'github-pages-deploy-action/m5icb7ze8'
/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
Completed deployment successfully! ✅
```

</details>

## Additional Comments

Workflow file for that commit: https://github.com/typegoose/typegoose/blob/4be023058b352fde261f7ebb6a6a02788d8a877e/.github/workflows/website.yml
Resulting commit: https://github.com/typegoose/typegoose/commit/1750a725eb1ac0d6b9060a208608457a85928561

Action version: 4.1.5

**Comment by JamesIves** [comment]
It looks like it's not handling the empty string case correctly [here](https://github.com/JamesIves/github-pages-deploy-action/blob/dev/src/constants.ts#L98-L108) as it's assumed this would never want to be empty. Right now if an email is not provided by the workflow then it will default to the one provided to the action, or a fallback noreply email. Can you help me understand why you'd want this to be empty?

**Comment by hasezoey** [comment]
> Can you help me understand why you'd want this to be empty?

i would like to not have the original commiter's email, to have a consistent gh-pages commit log (and because they did not actually make that commit) and because it is also not signed (which i would also not want to set-up for a workflow) and so i would like to not associate that commit with the original commiter's account

**Comment by YashdalfTheGray** [comment]
This is something that I'm thinking about using to push my own website which is a github pages website. If no one has taken a shot at this issue yet, I'd like to take a shot.

**Comment by JamesIves** [comment]
Apologies for the delay on this. This can be tested by pointing your workflow to the following: `JamesIves/github-pages-deploy-action@releases/v4-email-fix` and setting `git-config-email: ''`.

I'm aiming to bundle this with the next release.

**Comment by JamesIves** [comment]
This has been released as part of version `4.1.6`.

**Comment by JamesIves** [comment]
This has been changed in `4.1.7` slightly, you must now use `git-config-email: <>` in the field.
